src/main/java/org/skriptlang/skript/bukkit/potion/PotionModule.java

```diff
@@ -0,0 +1,208 @@
+			.name("Entity Potion Effect Event Action")
+			.description("Represents the action being performed in an 'entity potion effect' event.",
+				"'added' indicates the entity does not already have a potion effect of the event potion effect type.",
+				"'changed' indicates the entity already has a potion effect of the event potion effect type, but some property about the potion effect is changing.",
+				"'cleared' indicates that the effect is being removed because all of the entity's effects are being removed.",
+				"'removed' indicates that the event potion effect type has been specifically removed from the entity.")
+			.since("INSERT VERSION"));
+
+		// Added in 1.21
+		if (Skript.classExists("org.bukkit.potion.PotionEffectTypeCategory")) {
+			Classes.registerClass(new EnumClassInfo<>(PotionEffectTypeCategory.class, "potioneffecttypecategory", "potion effect type categories")
+				.user("potion ?effect ?type? category?(ies)?")
+				.name("Potion Effect Type Category")
+				.description("Represents the type of effect a potion effect type has on an entity.")
+				.since("INSERT VERSION"));
+			Comparators.registerComparator(PotionEffectType.class, PotionEffectTypeCategory.class,
+				(type, category) -> Relation.get(type.getCategory() == category));
+		}
+
+		// SkriptPotionEffect -> PotionEffect
+		Converters.registerConverter(SkriptPotionEffect.class, PotionEffect.class, SkriptPotionEffect::toPotionEffect);
+		// PotionEffect -> SkriptPotionEffect
+		Converters.registerConverter(PotionEffect.class, SkriptPotionEffect.class, SkriptPotionEffect::fromBukkitEffect);
+		// PotionEffectType -> SkriptPotionEffect
+		Converters.registerConverter(PotionEffectType.class, SkriptPotionEffect.class, SkriptPotionEffect::fromType);
+		// SkriptPotionEffect -> PotionEffectType
+		Converters.registerConverter(SkriptPotionEffect.class, PotionEffectType.class, SkriptPotionEffect::potionEffectType);
+	}
+
+	@Override
+	public void load(SkriptAddon addon) {
+		// Load Syntax
+		SyntaxRegistry registry = addon.syntaxRegistry();
+		// conditions
+		CondHasPotion.register(registry);
+		CondIsPoisoned.register(registry);
+		CondIsPotionAmbient.register(registry);
+		CondIsPotionInfinite.register(registry);
+		CondIsPotionInstant.register(registry);
+		CondPotionHasIcon.register(registry);
+		CondPotionHasParticles.register(registry);
+		// effects
+		EffApplyPotionEffect.register(registry);
+		EffPoison.register(registry);
+		EffPotionAmbient.register(registry);
+		EffPotionIcon.register(registry);
+		EffPotionInfinite.register(registry);
+		EffPotionParticles.register(registry);
+		// events
+		EvtEntityPotion.register(registry);
+		// expressions
+		ExprPotionAmplifier.register(registry);
+		ExprPotionDuration.register(registry);
+		ExprPotionEffect.register(registry);
+		ExprPotionEffects.register(registry);
+		ExprPotionEffectTypeCategory.register(registry);
+		ExprSecPotionEffect.register(registry);
+		ExprSkriptPotionEffect.register(registry);
```

Comment:
I like this new(?) registry pattern that was introduced, over the static initializers it used to be